File: QuanLyHienMauApi/QuanLyHienMauApi/Models/XacNhanHienMau.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyHienMauApi.Models
{
    public class XacNhanHienMau
    {
        [Key]
        public int ID { get; set; }

        public int? YeuCauID { get; set; }  

        [Required]
        public int NguoiHienID { get; set; }

        public int TrangThaiConfirm { get; set; } = 0;

        public DateTime NgayXacNhan { get; set; }
        public string? QRCode { get; set; }
        public int? ChieuCao { get; set; }      
        public double? CanNang { get; set; }    
        public int? HuyetApTamThu { get; set; } 
        public int? HuyetApTamTruong { get; set; }
        public string? NhomMauXacNhan { get; set; } 
        public string? RhXacNhan { get; set; }  
        public int? LuongMau { get; set; }

        [ForeignKey("YeuCauID")]
        public virtual YeuCauMau? YeuCau { get; set; }

        [ForeignKey("NguoiHienID")]
        public virtual NguoiDung? NguoiHien { get; set; }
    }
}
    

File: HienMau/app/src/main/java/com/example/hienmau/models/XacNhanHienMau.java
package com.example.hienmau.models;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class XacNhanHienMau implements Serializable {
    @SerializedName("ID")
    public int id;
    @SerializedName("YeuCauID")
    public int yeuCauId;
    @SerializedName("NguoiHienID")
    public int nguoiHienId;
    @SerializedName("TrangThaiConfirm")
    public int trangThaiConfirm;
    @SerializedName("NgayXacNhan")
    public String ngayXacNhan;
    @SerializedName("QRCode")
    public String qrCode;
    @SerializedName("ChieuCao")
    public Integer chieuCao;

    @SerializedName("CanNang")
    public Double canNang;

    @SerializedName("HuyetApTamThu")
    public Integer huyetApTamThu;

    @SerializedName("HuyetApTamTruong")
    public Integer huyetApTamTruong;
    @SerializedName("NhomMauXacNhan")
    public String nhomMauXacNhan;

    @SerializedName("RhXacNhan")
    public String rhXacNhan;

    @SerializedName("LuongMau")
    public int luongMau;
    @SerializedName("YeuCau")
    public YeuCauMau yeuCau;
    @SerializedName("NguoiHien")
    public NguoiDung nguoiHien;

    public int tempCount = 0;
}
